### PR TITLE
ULMS-376: external auth library

### PR DESCRIPTION
**Pull request by renskiy**
https://burning-heart.atlassian.net/browse/ULMS-376

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+type contextKey string
+
+const tokenContextKey = contextKey("token")
+
+// ErrAuthorization indicates 403 Forbidden HTTP status code
+var ErrAuthorization = fmt.Errorf("not authorized")
+
+// ErrTenant indicates tenant error
+var ErrTenant = fmt.Errorf("error performing tenant request, see server logs for details")
```
Comment (alexvassel) [inline auth.go]: why not `...tenant logs for details`?

File: client.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"net"
+	"net/http"
+	"time"
+)
+
+var client = &http.Client{
+	Timeout: 5 * time.Second,
```
Comment (alexvassel) [inline client.go]: magic number - 5. maybe store it in config?

File: auth.go (hunk 1 above)
Comment (renskiy) [reply auth.go]: removed

File: client.go (hunk 2 above)
Comment (renskiy) [reply client.go]: Maybe, but later. Just not necessary for now

**Pull request merged**